File: identhub/src/main/java/de/solarisbank/identhub/verfication/bank/VerificationBankViewModelFactory.java
package de.solarisbank.identhub.verfication.bank;

import androidx.lifecycle.ViewModel;

import de.solarisbank.identhub.domain.verification.bank.VerifyIBanUseCase;
import de.solarisbank.identhub.identity.IdentityModule;
import de.solarisbank.sdk.core.di.internal.Factory;
import de.solarisbank.sdk.core.di.internal.Provider;

public final class VerificationBankViewModelFactory implements Factory<ViewModel> {
    private final IdentityModule identityModule;
    private final Provider<VerifyIBanUseCase> verifyIBanUseCaseProvider;

    public VerificationBankViewModelFactory(IdentityModule identityModule, Provider<VerifyIBanUseCase> verifyIBanUseCaseProvider) {
        this.identityModule = identityModule;
        this.verifyIBanUseCaseProvider = verifyIBanUseCaseProvider;
    }

    public static VerificationBankViewModelFactory create(IdentityModule identityModule, Provider<VerifyIBanUseCase> verifyIBanUseCaseProvider) {
        return new VerificationBankViewModelFactory(identityModule, verifyIBanUseCaseProvider);
    }

    @Override
    public ViewModel get() {
        return identityModule.provideVerificationBankViewModel(verifyIBanUseCaseProvider.get());
    }
}

File: identhub/src/main/java/de/solarisbank/identhub/verfication/bank/VerificationBankViewModelFactory.kt
package de.solarisbank.identhub.verfication.bank

import androidx.lifecycle.SavedStateHandle
import androidx.lifecycle.ViewModel
import de.solarisbank.sdk.data.repository.SessionUrlRepository
import de.solarisbank.sdk.feature.config.InitializationInfoRepository
import de.solarisbank.sdk.feature.di.internal.Factory2
import de.solarisbank.sdk.feature.di.internal.Provider

class VerificationBankViewModelFactory(
    private val verificationBankModule: VerificationBankModule,
    private val sessionUrlRepositoryProvider: Provider<SessionUrlRepository>,
    private val initializationInfoRepositoryProvider: Provider<InitializationInfoRepository>
    ) : Factory2<ViewModel, SavedStateHandle> {

    override fun create(savedStateHandle: SavedStateHandle): ViewModel {
        return verificationBankModule.provideVerificationBankActivityViewModel(
            savedStateHandle,
            sessionUrlRepositoryProvider.get(),
            initializationInfoRepositoryProvider.get()
        )
    }
}